spec/support/time_matcher.rb

```diff
@@ -11,7 +11,11 @@
 end
 
 RSpec::Matchers.define :eq_time do |expected_time|
-  match { |actual_time| (actual_time - expected_time).abs < 0.001 }
+  match do |actual_time|
+    return true if expected_time.nil? && actual_time.nil?
+    return false if expected_time.nil? ^ actual_time.nil?
```

Comment:
Let's just say "no" to bitwise operators 😅 

```suggestion
    return false if expected_time.nil? || actual_time.nil?
```

Reply:
> Let's just say "no" to bitwise operators 😅

Why? :)

XOR is exactly what we need here. We test whether _only one_ of values is nil, and when both of them are not nill `||` returns not what we need:

```ruby
not_nil_value || not_nil_value == true # not what we need to test
not_nil_value ^ not_nil_value == false # what we need
```

Reply:
Just for the sake of simplicity 😅 We handle the case where the two are equal on the line above so here `||` and `^` are equivalent

Reply:
> We handle the case where the two are equal on the line above

Yeah, that's true, but my initial thinking was that `^`still better communicates the intention here. But `||` will work too, let's use it since it feels simpler 👍.